addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/SpringActuatorScanRuleTest.java

```diff
@@ -0,0 +1,328 @@
+        // Given
+        String path = "/";
+        this.nano.addHandler(
+                new NanoServerHandler(path) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        return newFixedLengthResponse(
+                                Response.Status.OK, NanoHTTPD.MIME_HTML, RELEVANT_BODY);
+                    }
+                });
+        HttpMessage msg = getHttpMessage(path);
+        rule.init(msg, parent);
+        this.nano.stop();
+        // When
+        rule.scan();
+        // Then = no exception
+        assertThat(alertsRaised, hasSize(0));
+        assertEquals(0, httpMessagesSent.size());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {ACTUATOR_CONTENT_TYPE, "Content-Type: application/json"})
+    void shouldAlertWhenScanMessageWithoutPathAndRelevantContent(String contentType)
+            throws HttpMalformedHeaderException {
+        // Given
+        String servePath = "";
+        String alertPath = REQ_PATH;
+        this.nano.addHandler(
+                new NanoServerHandler("/" + alertPath) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        return newFixedLengthResponse(
+                                Response.Status.OK, contentType, RELEVANT_BODY);
+                    }
+                });
+        HttpMessage msg = this.getHttpMessage(servePath);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        assertEquals(httpMessagesSent.size(), 1);
+        Alert alert = alertsRaised.get(0);
+        assertEquals(Alert.RISK_MEDIUM, alert.getRisk());
+        assertEquals(Alert.CONFIDENCE_MEDIUM, alert.getConfidence());
+        assertEquals(rule.getName(), alert.getName());
+        assertEquals(rule.getWascId(), alert.getWascId());
+    }
+
+    @Test
+    void shouldSendGetRequestWhenOriginalRequestWasNotGet() throws HttpMalformedHeaderException {
+        // Given
+        String path = "";
+        HttpMessage msg = getHttpMessage(path);
+        msg.getRequestHeader().setMethod(HttpRequestHeader.POST);
+        msg.getRequestHeader().addHeader(HttpHeader.CONTENT_TYPE, ACTUATOR_CONTENT_TYPE);
+        msg.setRequestBody("field1=value1&field2=value2");
+        rule.init(msg, parent);
+
+        // When
+        rule.scan();
+        // Then
+        assertEquals(httpMessagesSent.size(), 2);
+        assertEquals(httpMessagesSent.get(0).getRequestHeader().getMethod(), HttpRequestHeader.GET);
+        // assertNull(httpMessagesSent.get(0).getRequestHeader().getHeader(HttpHeader.CONTENT_TYPE));
+        assertEquals(httpMessagesSent.get(0).getRequestBody().length(), 0);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {ACTUATOR_CONTENT_TYPE, "Content-Type: application/json"})
```

Comment:
Had you planned other types? There's no point parameterizing for a single.

Reply:
There are others below as well. `shouldAlertIfTestedNestedUrlRespondsOKWithRelevantHeader`

Reply:
Uh actually I don't see contentType being used in the test method at all...

Reply:
And above: `shouldAlertWhenScanMessageWithoutPathAndRelevantContent`

Reply:
There are two types on Line 192, the Actuator Content Type and application/json - I might put them back up top

Reply:
I'm bit bit weak on how to make this work more elegantly, would it be an iterrable type and declare them up top or something else?
`private static final String ACTUATOR_CONTENT_TYPE[] =
            {"Content-Type: application/vnd.spring-boot.actuator.v2+json;charset=UTF-8", "Content-Type: applicaiton/json"};`

Reply:
Yup, you're right. I was reading it as a pair (expected/actual) or assignment for some reason... All good

Reply:
Don't worry about it, what you've done is fine. It was my reading/skimming of it that failed 😞